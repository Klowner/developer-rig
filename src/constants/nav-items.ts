import { ExtensionMode } from './extension-coordinator';

export const Labels: { [key: string]: string } = {
  ExtensionViews: 'Extension Views',
  ProductManagement: 'Product Management',
};
