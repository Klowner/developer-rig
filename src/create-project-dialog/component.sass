@import "../dialog-component/component.sass"
.project-dialog


.project-dialog__background
  @extend .dialog__backdrop

.project-dialog__dialog
  @extend .dialog
  height: 400px

.project-dialog__content
  @extend .container__left-block
  white-space: pre
  overflow-y: auto
  height: 250px
  width: 95%

.dialog__top-bar-container
  @extend .top-bar

.top-bar-container__title
  @extend .top-bar-element-left
  margin-top: 5px

.top-bar-container__escape
  @extend .top-bar-element-right
  width: $top-bar-dimension
  cursor: pointer
  width: 4em
  margin-top: 5px

.dialog__divider
  @extend .dialog__divider-width

.dialog__bottom-bar
  @extend .container__left-block

.bottom-bar__save
  @include bottom-bar-box(white, #7D55C7)

.bottom-bar__cancel
  @include bottom-bar-box(#7D55C7, white)
