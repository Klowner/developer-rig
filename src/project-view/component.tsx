import * as React from 'react';
import './component.sass';
import { RigProject } from '../core/models/rig';

type ProjectViewProps = RigProject & {
  edit: Function;
};

export class ProjectView extends React.Component<ProjectViewProps> {
  edit = () => {
    this.props.edit();
  }

  public render() {
    return (
      <div className="project-view__content">
        <div className="project-view__state-value">
          <label className="props-value__label"> Is Local </label>
          <input disabled={true} className="props-value__input" type="checkbox" name="isLocal" checked={this.props.isLocal} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Name </label>
          <input disabled={true} className="props-value__input" type="text" name="name" value={this.props.name} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Owner Name </label>
          <input disabled={true} className="props-value__input" type="text" name="ownerName" value={this.props.ownerName} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Client ID </label>
          <input disabled={true} className="props-value__input" type="text" name="clientId" value={this.props.clientId} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Version </label>
          <input disabled={true} className="props-value__input" type="text" name="version" value={this.props.version} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Front-end Directory </label>
          <input disabled={true} className="props-value__input" type="text" name="frontend" value={this.props.frontend} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Back-end File </label>
          <input disabled={true} className="props-value__input" type="text" name="backend" value={this.props.backend} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Secret </label>
          <input disabled={true} className="props-value__input" type="text" name="secret" value={this.props.secret} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Manifest </label>
          <textarea disabled={true} className="props-value__input" name="manifest" value={JSON.stringify(this.props.manifest)} />
        </div>
        <div className="project-view__state-value">
          <button onClick={this.edit}>Edit</button>
        </div>
      </div>
    );
  }
}
