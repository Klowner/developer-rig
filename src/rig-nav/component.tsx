import * as React from 'react';
import * as classNames from 'classnames';
import { Labels } from '../constants/nav-items';
import { UserDropdown } from '../user-dropdown';
import { LoginButton } from '../login-button';
import { UserSession } from '../core/models/user-session';
import { ExtensionManifest } from '../core/models/manifest';
import './component.sass';
import { MockApiDropdown } from './mock-api-dropdown';

export interface PublicProps {
  viewerHandler: Function,
  selectedView: string,
  error: string,
}

export interface ReduxStateProps {
  session?: UserSession,
  manifest?: ExtensionManifest,
  mockApiEnabled?: boolean,
}

type Props = PublicProps & ReduxStateProps;

export class RigNavComponent extends React.Component<Props> {
  private openProductManagementHandler = (): void => {
    const { session, manifest, viewerHandler } = this.props;
    if ((session && session.login) && (manifest && manifest.bitsEnabled)) {
      viewerHandler(Labels.ProductManagement);
    }
  }

  public render() {
    const { session, manifest, selectedView } = this.props;
    const projectOverviewClass = classNames({
      'offset': true,
      'top-nav-item': true,
      'top-nav-item__selected': selectedView === Labels.ProjectOverview,
    });
    const extensionViewsClass = classNames({
      'offset': true,
      'top-nav-item': true,
      'top-nav-item__selected': selectedView === Labels.ExtensionViews,
    });
    const productManagementClass = classNames({
      'top-nav-item': true,
      'top-nav-item__selected': selectedView === Labels.ProductManagement,
      'top-nav-item__disabled': !(session && session.login) || !(manifest && manifest.bitsEnabled),
    });

    if (this.props.error) {
      return (
        <div className='top-nav-error'>
          <a> {this.props.error} </a>
        </div>
      );
    } else {
      return (
        <div className='top-nav'>
          <div className='personal-bar'>
            <MockApiDropdown />
            {manifest && <div className='personal-bar__ext-name'>
              <span>{manifest.name}</span>
            </div>}
            <div className='top-nav-item__login'>
              {(session && session.login) ? <UserDropdown session={session} /> : <LoginButton />}
            </div>
          </div>
          <div className='top-nav__item-container'>
            <a className={projectOverviewClass} onClick={(event) => this.props.viewerHandler(Labels.ProjectOverview)}>
              Project Overview
            </a>
            <a className={extensionViewsClass} onClick={(event) => this.props.viewerHandler(Labels.ExtensionViews)}>
              Extension Views
            </a>
            <a className={productManagementClass} onClick={(event) => this.openProductManagementHandler()}>
              Manage Bits Products
            </a>
          </div>
        </div>
      );
    }
  }
}
