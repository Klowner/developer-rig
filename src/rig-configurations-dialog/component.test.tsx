import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { RigConfigurationsDialog } from './component';
import { createExtensionManifestForTest } from '../tests/constants/extension';

describe('<RigConfigurationsDialog />', () => {
  const setupShallow = setupShallowTest(RigConfigurationsDialog, () => ({
    config: createExtensionManifestForTest(),
    closeConfigurationsHandler: jest.fn(),
    refreshConfigurationsHandler: jest.fn()
  }));

  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('expect no config', () => {
    const { wrapper } = setupShallow({
      config: undefined,
    });
    expect(wrapper.find('.rig-configurations-view__content').text().trim()).toBe('');
  });

  it('should have some configurations', () => {
    const testConfig = createExtensionManifestForTest();
    const { wrapper } = setupShallow({
      config: testConfig
    });
    expect(JSON.parse(wrapper.find('.rig-configurations-view__content').text().trim())).toEqual(testConfig);
  });

  it('fires closeConfigurationsHandler when top exit button is clicked', () => {
    const { wrapper } = setupShallow();
    wrapper.find('.top-bar-container__escape').simulate('click');
    expect(wrapper.instance().props.closeConfigurationsHandler).toHaveBeenCalled();
  });

  it('fires closeConfigurationsHandler when cancel button is clicked', () => {
    const { wrapper } = setupShallow();
    wrapper.find('.bottom-bar__cancel').simulate('click');
    expect(wrapper.instance().props.closeConfigurationsHandler).toHaveBeenCalled();
  });

  it('fires refreshConfigurationsHandler when refresh button is clicked', () => {
    const { wrapper } = setupShallow();
    wrapper.find('.bottom-bar__refresh').simulate('click');
    expect(wrapper.instance().props.refreshConfigurationsHandler).toHaveBeenCalled();
  });
});
