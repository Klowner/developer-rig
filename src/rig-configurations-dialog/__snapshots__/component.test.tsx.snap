// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RigConfigurationsDialog /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RigConfigurationsDialog
    closeConfigurationsHandler={[Function]}
    config={
        Object {
            "authorName": "test",
            "bitsEnabled": true,
            "description": "test",
            "iconUrls": Object {
              "100x100": "test",
            },
            "id": "test",
            "name": "test",
            "requestIdentityLink": false,
            "sku": "test",
            "state": "test",
            "summary": "test",
            "vendorCode": "test",
            "version": "0.0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "size": 1024,
                "viewerUrl": "test",
                "zoom": false,
                "zoomPixels": 24,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [],
            "whitelistedPanelUrls": Array [],
          }
    }
    refreshConfigurationsHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="rig-configurations-view__background"
/>,
        <div
          className="rig-configurations-view__dialog"
>
          <div
                    className="dialog__top-bar-container"
          >
                    <div
                              className="top-bar-container__title"
                    >
                               Dev Rig Configurations 
                    </div>
                    <div
                              className="top-bar-container__escape"
                              onClick={[Function]}
                    >
                              <img
                                        alt="Close"
                                        src="close_icon.png"
                              />
                    </div>
          </div>
          <hr
                    className="dialog__divider"
          />
          <div
                    className="rig-configurations-view__content"
          >
                     
                    {
                      "authorName": "test",
                      "bitsEnabled": true,
                      "description": "test",
                      "iconUrls": {
                        "100x100": "test"
                      },
                      "id": "test",
                      "name": "test",
                      "requestIdentityLink": false,
                      "sku": "test",
                      "state": "test",
                      "summary": "test",
                      "vendorCode": "test",
                      "version": "0.0.1",
                      "views": {
                        "panel": {
                          "canLinkExternalContent": false,
                          "height": 300,
                          "viewerUrl": "test"
                        },
                        "config": {
                          "canLinkExternalContent": false,
                          "viewerUrl": "test"
                        },
                        "liveConfig": {
                          "canLinkExternalContent": false,
                          "viewerUrl": "test"
                        },
                        "component": {
                          "aspectHeight": 3000,
                          "aspectWidth": 2500,
                          "canLinkExternalContent": false,
                          "viewerUrl": "test",
                          "size": 1024,
                          "zoom": false,
                          "zoomPixels": 24
                        }
                      },
                      "whitelistedConfigUrls": [],
                      "whitelistedPanelUrls": []
                    }
                     
          </div>
          <hr
                    className="dialog__divider"
          />
          <div
                    className="dialog_bottom-bar"
          >
                    <div
                              className="bottom-bar__refresh"
                              onClick={[Function]}
                    >
                               Refresh 
                    </div>
                    <div
                              className="bottom-bar__cancel"
                              onClick={[Function]}
                    >
                               Cancel 
                    </div>
          </div>
</div>,
      ],
      "className": "rig-configurations-view",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "rig-configurations-view__background",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="dialog__top-bar-container"
>
              <div
                            className="top-bar-container__title"
              >
                             Dev Rig Configurations 
              </div>
              <div
                            className="top-bar-container__escape"
                            onClick={[Function]}
              >
                            <img
                                          alt="Close"
                                          src="close_icon.png"
                            />
              </div>
</div>,
            <hr
              className="dialog__divider"
/>,
            <div
              className="rig-configurations-view__content"
>
               
              {
                "authorName": "test",
                "bitsEnabled": true,
                "description": "test",
                "iconUrls": {
                  "100x100": "test"
                },
                "id": "test",
                "name": "test",
                "requestIdentityLink": false,
                "sku": "test",
                "state": "test",
                "summary": "test",
                "vendorCode": "test",
                "version": "0.0.1",
                "views": {
                  "panel": {
                    "canLinkExternalContent": false,
                    "height": 300,
                    "viewerUrl": "test"
                  },
                  "config": {
                    "canLinkExternalContent": false,
                    "viewerUrl": "test"
                  },
                  "liveConfig": {
                    "canLinkExternalContent": false,
                    "viewerUrl": "test"
                  },
                  "component": {
                    "aspectHeight": 3000,
                    "aspectWidth": 2500,
                    "canLinkExternalContent": false,
                    "viewerUrl": "test",
                    "size": 1024,
                    "zoom": false,
                    "zoomPixels": 24
                  }
                },
                "whitelistedConfigUrls": [],
                "whitelistedPanelUrls": []
              }
               
</div>,
            <hr
              className="dialog__divider"
/>,
            <div
              className="dialog_bottom-bar"
>
              <div
                            className="bottom-bar__refresh"
                            onClick={[Function]}
              >
                             Refresh 
              </div>
              <div
                            className="bottom-bar__cancel"
                            onClick={[Function]}
              >
                             Cancel 
              </div>
</div>,
          ],
          "className": "rig-configurations-view__dialog",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="top-bar-container__title"
>
                   Dev Rig Configurations 
</div>,
                <div
                  className="top-bar-container__escape"
                  onClick={[Function]}
>
                  <img
                                    alt="Close"
                                    src="close_icon.png"
                  />
</div>,
              ],
              "className": "dialog__top-bar-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Dev Rig Configurations ",
                  "className": "top-bar-container__title",
                },
                "ref": null,
                "rendered": " Dev Rig Configurations ",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="Close"
                    src="close_icon.png"
/>,
                  "className": "top-bar-container__escape",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Close",
                    "src": "close_icon.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "dialog__divider",
            },
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "{
  \\"authorName\\": \\"test\\",
  \\"bitsEnabled\\": true,
  \\"description\\": \\"test\\",
  \\"iconUrls\\": {
    \\"100x100\\": \\"test\\"
  },
  \\"id\\": \\"test\\",
  \\"name\\": \\"test\\",
  \\"requestIdentityLink\\": false,
  \\"sku\\": \\"test\\",
  \\"state\\": \\"test\\",
  \\"summary\\": \\"test\\",
  \\"vendorCode\\": \\"test\\",
  \\"version\\": \\"0.0.1\\",
  \\"views\\": {
    \\"panel\\": {
      \\"canLinkExternalContent\\": false,
      \\"height\\": 300,
      \\"viewerUrl\\": \\"test\\"
    },
    \\"config\\": {
      \\"canLinkExternalContent\\": false,
      \\"viewerUrl\\": \\"test\\"
    },
    \\"liveConfig\\": {
      \\"canLinkExternalContent\\": false,
      \\"viewerUrl\\": \\"test\\"
    },
    \\"component\\": {
      \\"aspectHeight\\": 3000,
      \\"aspectWidth\\": 2500,
      \\"canLinkExternalContent\\": false,
      \\"viewerUrl\\": \\"test\\",
      \\"size\\": 1024,
      \\"zoom\\": false,
      \\"zoomPixels\\": 24
    }
  },
  \\"whitelistedConfigUrls\\": [],
  \\"whitelistedPanelUrls\\": []
}",
                " ",
              ],
              "className": "rig-configurations-view__content",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "{
  \\"authorName\\": \\"test\\",
  \\"bitsEnabled\\": true,
  \\"description\\": \\"test\\",
  \\"iconUrls\\": {
    \\"100x100\\": \\"test\\"
  },
  \\"id\\": \\"test\\",
  \\"name\\": \\"test\\",
  \\"requestIdentityLink\\": false,
  \\"sku\\": \\"test\\",
  \\"state\\": \\"test\\",
  \\"summary\\": \\"test\\",
  \\"vendorCode\\": \\"test\\",
  \\"version\\": \\"0.0.1\\",
  \\"views\\": {
    \\"panel\\": {
      \\"canLinkExternalContent\\": false,
      \\"height\\": 300,
      \\"viewerUrl\\": \\"test\\"
    },
    \\"config\\": {
      \\"canLinkExternalContent\\": false,
      \\"viewerUrl\\": \\"test\\"
    },
    \\"liveConfig\\": {
      \\"canLinkExternalContent\\": false,
      \\"viewerUrl\\": \\"test\\"
    },
    \\"component\\": {
      \\"aspectHeight\\": 3000,
      \\"aspectWidth\\": 2500,
      \\"canLinkExternalContent\\": false,
      \\"viewerUrl\\": \\"test\\",
      \\"size\\": 1024,
      \\"zoom\\": false,
      \\"zoomPixels\\": 24
    }
  },
  \\"whitelistedConfigUrls\\": [],
  \\"whitelistedPanelUrls\\": []
}",
              " ",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "dialog__divider",
            },
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="bottom-bar__refresh"
                  onClick={[Function]}
>
                   Refresh 
</div>,
                <div
                  className="bottom-bar__cancel"
                  onClick={[Function]}
>
                   Cancel 
</div>,
              ],
              "className": "dialog_bottom-bar",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Refresh ",
                  "className": "bottom-bar__refresh",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": " Refresh ",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Cancel ",
                  "className": "bottom-bar__cancel",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": " Cancel ",
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="rig-configurations-view__background"
/>,
          <div
            className="rig-configurations-view__dialog"
>
            <div
                        className="dialog__top-bar-container"
            >
                        <div
                                    className="top-bar-container__title"
                        >
                                     Dev Rig Configurations 
                        </div>
                        <div
                                    className="top-bar-container__escape"
                                    onClick={[Function]}
                        >
                                    <img
                                                alt="Close"
                                                src="close_icon.png"
                                    />
                        </div>
            </div>
            <hr
                        className="dialog__divider"
            />
            <div
                        className="rig-configurations-view__content"
            >
                         
                        {
                          "authorName": "test",
                          "bitsEnabled": true,
                          "description": "test",
                          "iconUrls": {
                            "100x100": "test"
                          },
                          "id": "test",
                          "name": "test",
                          "requestIdentityLink": false,
                          "sku": "test",
                          "state": "test",
                          "summary": "test",
                          "vendorCode": "test",
                          "version": "0.0.1",
                          "views": {
                            "panel": {
                              "canLinkExternalContent": false,
                              "height": 300,
                              "viewerUrl": "test"
                            },
                            "config": {
                              "canLinkExternalContent": false,
                              "viewerUrl": "test"
                            },
                            "liveConfig": {
                              "canLinkExternalContent": false,
                              "viewerUrl": "test"
                            },
                            "component": {
                              "aspectHeight": 3000,
                              "aspectWidth": 2500,
                              "canLinkExternalContent": false,
                              "viewerUrl": "test",
                              "size": 1024,
                              "zoom": false,
                              "zoomPixels": 24
                            }
                          },
                          "whitelistedConfigUrls": [],
                          "whitelistedPanelUrls": []
                        }
                         
            </div>
            <hr
                        className="dialog__divider"
            />
            <div
                        className="dialog_bottom-bar"
            >
                        <div
                                    className="bottom-bar__refresh"
                                    onClick={[Function]}
                        >
                                     Refresh 
                        </div>
                        <div
                                    className="bottom-bar__cancel"
                                    onClick={[Function]}
                        >
                                     Cancel 
                        </div>
            </div>
</div>,
        ],
        "className": "rig-configurations-view",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "rig-configurations-view__background",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="dialog__top-bar-container"
>
                <div
                                className="top-bar-container__title"
                >
                                 Dev Rig Configurations 
                </div>
                <div
                                className="top-bar-container__escape"
                                onClick={[Function]}
                >
                                <img
                                                alt="Close"
                                                src="close_icon.png"
                                />
                </div>
</div>,
              <hr
                className="dialog__divider"
/>,
              <div
                className="rig-configurations-view__content"
>
                 
                {
                  "authorName": "test",
                  "bitsEnabled": true,
                  "description": "test",
                  "iconUrls": {
                    "100x100": "test"
                  },
                  "id": "test",
                  "name": "test",
                  "requestIdentityLink": false,
                  "sku": "test",
                  "state": "test",
                  "summary": "test",
                  "vendorCode": "test",
                  "version": "0.0.1",
                  "views": {
                    "panel": {
                      "canLinkExternalContent": false,
                      "height": 300,
                      "viewerUrl": "test"
                    },
                    "config": {
                      "canLinkExternalContent": false,
                      "viewerUrl": "test"
                    },
                    "liveConfig": {
                      "canLinkExternalContent": false,
                      "viewerUrl": "test"
                    },
                    "component": {
                      "aspectHeight": 3000,
                      "aspectWidth": 2500,
                      "canLinkExternalContent": false,
                      "viewerUrl": "test",
                      "size": 1024,
                      "zoom": false,
                      "zoomPixels": 24
                    }
                  },
                  "whitelistedConfigUrls": [],
                  "whitelistedPanelUrls": []
                }
                 
</div>,
              <hr
                className="dialog__divider"
/>,
              <div
                className="dialog_bottom-bar"
>
                <div
                                className="bottom-bar__refresh"
                                onClick={[Function]}
                >
                                 Refresh 
                </div>
                <div
                                className="bottom-bar__cancel"
                                onClick={[Function]}
                >
                                 Cancel 
                </div>
</div>,
            ],
            "className": "rig-configurations-view__dialog",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="top-bar-container__title"
>
                     Dev Rig Configurations 
</div>,
                  <div
                    className="top-bar-container__escape"
                    onClick={[Function]}
>
                    <img
                                        alt="Close"
                                        src="close_icon.png"
                    />
</div>,
                ],
                "className": "dialog__top-bar-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Dev Rig Configurations ",
                    "className": "top-bar-container__title",
                  },
                  "ref": null,
                  "rendered": " Dev Rig Configurations ",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="Close"
                      src="close_icon.png"
/>,
                    "className": "top-bar-container__escape",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Close",
                      "src": "close_icon.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "dialog__divider",
              },
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "{
  \\"authorName\\": \\"test\\",
  \\"bitsEnabled\\": true,
  \\"description\\": \\"test\\",
  \\"iconUrls\\": {
    \\"100x100\\": \\"test\\"
  },
  \\"id\\": \\"test\\",
  \\"name\\": \\"test\\",
  \\"requestIdentityLink\\": false,
  \\"sku\\": \\"test\\",
  \\"state\\": \\"test\\",
  \\"summary\\": \\"test\\",
  \\"vendorCode\\": \\"test\\",
  \\"version\\": \\"0.0.1\\",
  \\"views\\": {
    \\"panel\\": {
      \\"canLinkExternalContent\\": false,
      \\"height\\": 300,
      \\"viewerUrl\\": \\"test\\"
    },
    \\"config\\": {
      \\"canLinkExternalContent\\": false,
      \\"viewerUrl\\": \\"test\\"
    },
    \\"liveConfig\\": {
      \\"canLinkExternalContent\\": false,
      \\"viewerUrl\\": \\"test\\"
    },
    \\"component\\": {
      \\"aspectHeight\\": 3000,
      \\"aspectWidth\\": 2500,
      \\"canLinkExternalContent\\": false,
      \\"viewerUrl\\": \\"test\\",
      \\"size\\": 1024,
      \\"zoom\\": false,
      \\"zoomPixels\\": 24
    }
  },
  \\"whitelistedConfigUrls\\": [],
  \\"whitelistedPanelUrls\\": []
}",
                  " ",
                ],
                "className": "rig-configurations-view__content",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "{
  \\"authorName\\": \\"test\\",
  \\"bitsEnabled\\": true,
  \\"description\\": \\"test\\",
  \\"iconUrls\\": {
    \\"100x100\\": \\"test\\"
  },
  \\"id\\": \\"test\\",
  \\"name\\": \\"test\\",
  \\"requestIdentityLink\\": false,
  \\"sku\\": \\"test\\",
  \\"state\\": \\"test\\",
  \\"summary\\": \\"test\\",
  \\"vendorCode\\": \\"test\\",
  \\"version\\": \\"0.0.1\\",
  \\"views\\": {
    \\"panel\\": {
      \\"canLinkExternalContent\\": false,
      \\"height\\": 300,
      \\"viewerUrl\\": \\"test\\"
    },
    \\"config\\": {
      \\"canLinkExternalContent\\": false,
      \\"viewerUrl\\": \\"test\\"
    },
    \\"liveConfig\\": {
      \\"canLinkExternalContent\\": false,
      \\"viewerUrl\\": \\"test\\"
    },
    \\"component\\": {
      \\"aspectHeight\\": 3000,
      \\"aspectWidth\\": 2500,
      \\"canLinkExternalContent\\": false,
      \\"viewerUrl\\": \\"test\\",
      \\"size\\": 1024,
      \\"zoom\\": false,
      \\"zoomPixels\\": 24
    }
  },
  \\"whitelistedConfigUrls\\": [],
  \\"whitelistedPanelUrls\\": []
}",
                " ",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "dialog__divider",
              },
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="bottom-bar__refresh"
                    onClick={[Function]}
>
                     Refresh 
</div>,
                  <div
                    className="bottom-bar__cancel"
                    onClick={[Function]}
>
                     Cancel 
</div>,
                ],
                "className": "dialog_bottom-bar",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Refresh ",
                    "className": "bottom-bar__refresh",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": " Refresh ",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Cancel ",
                    "className": "bottom-bar__cancel",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": " Cancel ",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
